lms/djangoapps/courseware/module_render.py

```diff
@@ -917,7 +918,8 @@ def get_module_by_usage_id(request, course_id, usage_id, disable_staff_debug_inf
     field_data_cache = FieldDataCache.cache_for_descriptor_descendents(
         course_id,
         user,
-        descriptor
+        descriptor,
+        read_only=CrawlersConfig.is_crawler(request),
```

Comment:
**Question:** Why were only these 2 particular factory calls for the `FieldDataCache` updated?  What about other places that create `FieldDataCache`?  Alternatively, the check for Crawlers could be made inside `FieldDataCache` itself, unless that breaks an abstraction barrier that you had in mind.